src/main/java/net/openhft/chronicle/queue/impl/single/TableDirectoryListing.java

```diff
@@ -64,16 +75,24 @@ final class TableDirectoryListing extends AbstractCloseable implements Directory
     public void init() {
         throwExceptionIfClosedInSetter();
 
-        tableStore.doWithExclusiveLock(ts -> {
-            maxCycleValue = ts.acquireValueFor(HIGHEST_CREATED_CYCLE);
-            minCycleValue = ts.acquireValueFor(LOWEST_CREATED_CYCLE);
-            minCycleValue.compareAndSwapValue(Long.MIN_VALUE, UNSET_MIN_CYCLE);
-            modCount = ts.acquireValueFor(MOD_COUNT);
-            if (modCount.getVolatileValue() == Long.MIN_VALUE) {
-                modCount.compareAndSwapValue(Long.MIN_VALUE, 0);
-            }
-            return this;
-        });
+        final Runnable initLongValues = () -> {
+            maxCycleValue = tableStore.acquireValueFor(HIGHEST_CREATED_CYCLE);
+            minCycleValue = tableStore.acquireValueFor(LOWEST_CREATED_CYCLE);
+            modCount = tableStore.acquireValueFor(MOD_COUNT);
+        };
+
+        if (readOnly) {
+            initLongValues.run();
```

Comment:
if we are racing with a writeable TableDirectoryListing then it is possible this may fail

Reply:
We'll just need a version of `TableStore#acquireValue` that doesn't create it if it doesn't exist. Looking at the code that doesn't seem that tricky?

Reply:
Ah I see you throw now if it's read-only and doesn't exist. Nice.